tools/codegen/gen.py

```diff
@@ -104,6 +104,38 @@ def cpp_string(s: str) -> str:
 # to be generated.  This pattern makes it convenient to use map, concatMap
 # and similar functional combinators.
 
+def static_dispatch_extra_headers(backends: Optional[List[DispatchKey]]) -> str:
+    if not backends:
+        return ''
+    return '\n'.join(f'#include <ATen/{key}Functions.h>' for key in backends) + """
+#include <ATen/DefaultBackendFunctions.h>
+#include <ATen/MathFunctions.h>
+"""
+
+def static_dispatch(
+    f: NativeFunction, cpp_sig: CppSignature,
+    *, method: bool, backends: Optional[List[DispatchKey]]
+) -> Optional[str]:
+    if not backends or f.manual_kernel_registration:
+        return None
+
+    assert len(backends) <= 1, "Multi-backend static dispatch is not yet supported!"
```

Comment:
just curious, is there something preventing multi-backend static dispatch from working? It would probably require extra CI jobs, so I could imagine not bothering with it until there's a need for it.

Reply:
oh yeah, nvm. I guess only certain combinations would be valid (CPU and QuantizedCPU) and there's not much point sorting through it unless there's a use case

Reply:
@bdhirsh I think the biggest reason to support single-backend dispatch is that only under that constraint can we completely dispense with dispatch key computation, as opposed to hoisting it out of the dispatcher into the generated wrappers. 

It's true that we may get use cases for the multi-key version of this, but I don't know that we need to build it out now - and if/when it happens, I think we'd definitely want to preserve the single-key version as a distinct codegen path.